### DISCUSSION TITLE
Firmware Setup

### DISCUSSION THREAD

**Discussion by FelixHauser** [jasoncoon/esp8266-fastled-webserver#255] [General]
Hi there,

I have extensively used an older version of the fastled-webserved in the past. I build Nanoleaf replicas and name signs, etc. It was a painless setup on the former .ino file, where I could setup what kind of setup I had (Nanoleaf, signs, etc.), the number of LEDs, the pin they were attached to, the device name and some other stuff. Never had a problem with it.

I am trying now to set up the most recent version. I have compiled the sketch, I have uploaded to my board (ESP8266), I have uploaded the webdata. No problem. However I fail to see the whole setup parameters I could setup before. Since I did not see them on the sketch, I assumed I could setup everything on the web interface. However this is not the case. I haven' found any documentation about it on the repository or discussions.

So, I guess that's my question. How do I setup all those things on the current version of the webserver?

Thanks in advance.

**Comment by jasoncoon** [comment]
Hey @FelixHauser,

A while back we combined all of the various product branches into a single one. It's now all configured with preprocessor directives, conditional includes, etc.  We also added support for PlatformIO which, if you haven't tried it yet, is a huge step up from the Arduino IDE.

More info: https://github.com/jasoncoon/esp8266-fastled-webserver/discussions/206#discussioncomment-1719628

**Reply by FelixHauser** [reply]
thanks for your reply @jasoncoon 

unfortunately however, it hasn't become more clear to me. I am able to download a sketch or Platformio project, modify a couple of variables and upload it to the board, but not much more. I do not understand what you mean by "preprocessor directives, conditional includes, etc.". Could you point at the files I have to tinker with? it would be really appreciated!


thanks!

**Comment by qawsed55** [comment]
@FelixHauser in Platformio.ini you need to setup `default_envs` (what type of version you want to compile)
in `esp8266-fastled-webserver\include\configs\controller` you can setup all you need 
or you can setup  `\platformio_override_example.ini` and rename it to `\platformio_override.ini` and setup all there
In down (blue) part of PIO window press on Default (esp8266-fastled...) and select what type of product you want to compile. 
Here, also you can find other usefull comands.
